packages/block-library/src/navigation/edit/index.js

```diff
@@ -704,20 +659,24 @@ function Navigation( {
 		<EntityProvider kind="postType" type="wp_navigation" id={ ref }>
 			<RecursionProvider uniqueId={ recursionId }>
 				<BlockControls>
-					{ ! isDraftNavigationMenu && isEntityAvailable && (
```

Comment:
I don't think these conditions do anything important, because there are already early returns in the component for missing entities or unselected menus.